.github/workflows/build_upload_pypi_wheels.yml

```diff
@@ -111,9 +137,67 @@ jobs:
             path: wheelhouse/*-${{ matrix.wheelname }}*_${{ matrix.cibw_archs }}.whl
             if-no-files-found: error
 
+  test-sdist:
+    needs: build-sdist
+    name: Test build from sdist on Windows
+    runs-on: windows-latest
+    steps:
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+
+      - name: Find MSVC and set environment variables
+        shell: bash -l {0}
+        env:
+          MSVC_PREFIX: 'C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC'
+        run: |
+          echo "Available MSVC installations:"
+          ls "$MSVC_PREFIX"
+
+          MSVC_BIN=$(ls "$MSVC_PREFIX" | tail -n 1)\\bin\\HostX64\\x64
+          CC="$MSVC_PREFIX\\$MSVC_BIN\\cl.exe"
+          echo "CC: $CC"
+          echo "CC=$CC" >> $GITHUB_ENV
+
+          CC_LD="$MSVC_PREFIX\\$MSVC_BIN\\link.exe"
+          echo "CC_LD: $CC_LD"
+          echo "CC_LD=$CC_LD" >> $GITHUB_ENV
+
+      - name: Download sdist
+        uses: actions/download-artifact@v4
+        with:
+          path: dist/
+          merge-multiple: true
+
+      - name: List downloaded sdist
+        run: |
+          ls -R dist/
+
+      - name: Update pip
+        shell: bash -l {0}
+        run: |
+          python -m pip install --upgrade pip
+
+      - name: Install from sdist
+        shell: bash -l {0}
+        run: python -m pip install $(find dist -name 'euphonic-*.tar.gz')[matplotlib,phonopy_reader,brille,test]
```

Comment:
Unlikely to be multiple, but what happens if there are?

Reply:
If they are different we should get a Pip resolution error

Reply:
We could have something like

```bash
SDIST=$(find dist -name 'euphonic-*.tar.gz')
if test (echo $SDIST | wc -w) -ne 1
  then echo "Found $SDIST : not a single sdist file"; exit 1
fi
python -m pip install ${SDIST}[matplotlib,phonopy_reader,brille,test]
```

but I'm not sure it would be a lot safer.